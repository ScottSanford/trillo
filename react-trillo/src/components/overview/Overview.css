/* HOTEL OVERVIEW */
.overview {
    align-items: center;
    border-bottom: var(--line);
    display: flex;
}

.overview__heading {
    font-size: 2.25rem;
    font-weight: 300;
    text-transform: uppercase;
    letter-spacing: 1px;
    padding: 1.5rem 3rem;
}

@media only screen and (max-width: 37.5em) {
    .overview__heading {
        font-size: 1.8rem;
        padding: 1.25rem 2rem;
    }
}

.overview__stars {
    display: flex;
    flex: 1;
    margin-right: auto;
}
.overview__icon-star,
.overview__icon-location {
    fill: var(--color-primary);
    height: 1.75rem;
    width: 1.75rem;
}

.overview__location {
    align-items: center;
    display: flex;
    font-size: 1.2rem;
}

.overview__icon-location {
    margin-right: 0.5rem;
}

.overview__rating {
    align-self: stretch;
    background-color: var(--color-primary);
    color: var(--color-white);
    margin-left: 3rem;
    padding: 0 2.25rem;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
}

@media only screen and (max-width: 37.5em) {
    .overview__rating {
        padding: 0 1.5rem;
    }
}

.overview__rating-average {
    font-size: 2.25rem;
    font-weight: 300;
    margin-bottom: -3px;
}

@media only screen and (max-width: 37.5em) {
    .overview__rating-average {
        font-size: 1.8rem;
    }
}
.overview__rating-count {
    font-size: 0.8rem;
    text-transform: uppercase;
}

@media only screen and (max-width: 37.5em) {
    .overview__rating-count {
        font-size: 0.5rem;
    }
}