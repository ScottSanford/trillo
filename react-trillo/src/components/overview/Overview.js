import React from 'react'
import './Overview.css'

export default function Overview({ hotel }) {

  const { name, location, overallRatingAvg, overallRatingCount } = hotel

	return (
		<div className="overview">
          <h1 className="overview__heading">{name}</h1>
          <div className="overview__stars">
            <svg className="overview__icon-star">
              <use xlinkHref="img/sprite.svg#icon-star" />
            </svg>
            <svg className="overview__icon-star">
              <use xlinkHref="img/sprite.svg#icon-star" />
            </svg>
            <svg className="overview__icon-star">
              <use xlinkHref="img/sprite.svg#icon-star" />
            </svg>
            <svg className="overview__icon-star">
              <use xlinkHref="img/sprite.svg#icon-star" />
            </svg>
            <svg className="overview__icon-star">
              <use xlinkHref="img/sprite.svg#icon-star" />
            </svg>
          </div>
          <div className="overview__location">
            <svg className="overview__icon-location">
              <use xlinkHref="img/sprite.svg#icon-location-pin" />
            </svg>
            <button className="btn-inline">{location}</button>
          </div>

          <div className="overview__rating">
            <div className="overview__rating-average">{overallRatingAvg}</div>
            <div className="overview__rating-count">{overallRatingCount} votes</div>
          </div>
        </div>
	)
}