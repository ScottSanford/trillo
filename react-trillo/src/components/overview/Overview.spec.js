import React from 'react'
import { shallow } from 'enzyme'
import Overview from './Overview'

describe('Overview Component', () => {

	const minProps = {
		hotel: {}
	}

	it('should match snapshot', () => {
		const wrapper = shallow(<Overview {...minProps} />)
		expect(wrapper).toMatchSnapshot()
	})
})