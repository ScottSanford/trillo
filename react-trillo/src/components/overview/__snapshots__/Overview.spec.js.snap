// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Overview Component should match snapshot 1`] = `
<div
  className="overview"
>
  <h1
    className="overview__heading"
  />
  <div
    className="overview__stars"
  >
    <svg
      className="overview__icon-star"
    >
      <use
        xlinkHref="img/sprite.svg#icon-star"
      />
    </svg>
    <svg
      className="overview__icon-star"
    >
      <use
        xlinkHref="img/sprite.svg#icon-star"
      />
    </svg>
    <svg
      className="overview__icon-star"
    >
      <use
        xlinkHref="img/sprite.svg#icon-star"
      />
    </svg>
    <svg
      className="overview__icon-star"
    >
      <use
        xlinkHref="img/sprite.svg#icon-star"
      />
    </svg>
    <svg
      className="overview__icon-star"
    >
      <use
        xlinkHref="img/sprite.svg#icon-star"
      />
    </svg>
  </div>
  <div
    className="overview__location"
  >
    <svg
      className="overview__icon-location"
    >
      <use
        xlinkHref="img/sprite.svg#icon-location-pin"
      />
    </svg>
    <button
      className="btn-inline"
    />
  </div>
  <div
    className="overview__rating"
  >
    <div
      className="overview__rating-average"
    />
    <div
      className="overview__rating-count"
    >
       votes
    </div>
  </div>
</div>
`;
